### PR TITLE
[WIP]Dev/xygu/layout provider debug utils

### PR DESCRIPTION
**Pull request by XiaotianNetlift**
GitHub Issue (If applicable): n/a

## PR Type
What kind of change does this PR introduce?
Feature

## What is the current behavior?
It is hard/slow to debug android layout.

## What is the new behavior?
- ~~You can use `LayoutProvider.DebugLayoutChanged` event to debug the layout calculation with the same context as in `LayoutProvider` without having to recompile `Uno.UI` and your test app.~~
- There is a new linqpad script to help visualize android layout in real-time.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable): n/a

### CODE REVIEW

**Comment by XiaotianNetlift** [comment]
@XiaotianNetlift todo:
- [ ] screenshot
- [ ] linqpad script
  - [ ] add fix for console-color
  - [ ] clean it up

File: utils/script/layout visualizer.linq [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+
+#define CLEAR_OLD_LOG  // run `adb logcat -c` before starting to parse log
+#define ENABLE_RAW_INPUT_FILTER // filter out noisy system logs
+#define REMOVE_DUPLICATED_NEWLINES
+//#define DISABLE_SCREENSHOT // on slower device, you may want to uncomment this
+
+void Main()
+{
+	// This script uses adb to obtain the layout data logged by LayoutProvider.
+	// Please ensure that `adb` is installed on this machine and is added to envinronment path.
```
Comment (MatFillion) [inline utils/script/layout visualizer.linq]:
```suggestion
	// Please ensure that `adb` is installed on this machine and is added to environment path.
```

File: utils/script/layout visualizer.linq [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+  <Namespace>Newtonsoft.Json</Namespace>
+</Query>
+
+#define CLEAR_OLD_LOG  // run `adb logcat -c` before starting to parse log
+#define ENABLE_RAW_INPUT_FILTER // filter out noisy system logs
+#define REMOVE_DUPLICATED_NEWLINES
+//#define DISABLE_SCREENSHOT // on slower device, you may want to uncomment this
+
+void Main()
+{
```
Comment (agneszitte-nventive) [inline utils/script/layout visualizer.linq]: You can add a comment that precise to have SoftInput.AdjustNothing ;)

**Comment by XiaotianNetlift** [comment]
replaced by #1294

**Pull request closed without merging**